android/TodokTodok/app/src/main/java/com/team/todoktodok/presentation/view/discussiondetail/commentdetail/vm/CommentDetailViewModelFactory.kt

```
@@ -0,0 +1,27 @@
+    override fun <T : ViewModel> create(
+        modelClass: Class<T>,
+        extras: CreationExtras,
+    ): T {
+        val savedStateHandle = extras.createSavedStateHandle()
+        if (modelClass.isAssignableFrom(CommentDetailViewModel::class.java)) {
+            return CommentDetailViewModel(
+                savedStateHandle,
+                commentRepository,
+            ) as T
+        } else {
+            throw IllegalArgumentException()
```

detekt (1.23.8)

[warning] A call to the default constructor of an exception was detected. Instead one of the constructor overloads should be called. This allows to provide more meaningful exceptions.

(detekt.exceptions.ThrowingExceptionsWithoutMessageOrCause)